src/main/java/seedu/address/logic/commands/contact/AddCommand.java

```diff
@@ -72,13 +72,23 @@ public AddCommand(Name name, Phone phone, Email email, Address address, Set<Tag>
         this.remark = remark;
     }
 
+    private int getNewPersonId(Model model) {
+        List<Person> allPersons = model.getAllPersons();
+        int maxIdValue = 0;
+        for (Person p: allPersons) {
+            if (p.getId() > maxIdValue) {
+                maxIdValue = p.getId();
+            }
+        }
+        return maxIdValue + 1;
+    }
```

Comment:
I think this will be get very costly. Is there an alternative way to do this? like a static counter?

Reply:
I think the counter resets since a new `AddressBook` instance is initialised every time the app starts, so if we do a static variable we might have to store it into the storage file.

Reply:
how about creating a static method doing this, then do private static final counter = methodcall()? then you call once every time you boot up instead of every insertion?

Reply:
Might need some time to think about how to do this. Maybe put this under 1.3b? Need to work on other features.

Reply:
I think min yips PR of list all sales has done this part